src/platform/platform.c

```diff
@@ -320,37 +320,66 @@ static int get_current_dir_filepath(int32_t buffer_len, char_t* buffer, int32_t
 }
 
 // Globals to hold hostfxr exports
+
+static hostfxr_initialize_for_dotnet_command_line_fn init_self_contained_fptr;
 static hostfxr_initialize_for_runtime_config_fn init_fptr;
 static hostfxr_get_runtime_delegate_fn get_delegate_fptr;
 static hostfxr_close_fn close_fptr;
 
-static void load_hostfxr(void)
+static void load_hostfxr(const char_t* assembly_path)
 {
     // Discover the path to hostfxr.
     char_t buffer[DNNE_MAX_PATH];
     size_t buffer_size = DNNE_ARRAY_SIZE(buffer);
-    int rc = get_hostfxr_path(buffer, &buffer_size, NULL);
+    struct get_hostfxr_parameters params;
+    params.size = sizeof(params);
+    params.assembly_path = assembly_path;
+    params.dotnet_root = NULL;
+    int rc = get_hostfxr_path(buffer, &buffer_size, &params);
     if (is_failure(rc))
         noreturn_runtime_load_failure(rc);
 
     // Load hostfxr and get desired exports.
     void* lib = load_library(buffer);
+    init_self_contained_fptr = (hostfxr_initialize_for_dotnet_command_line_fn)get_export(lib, "hostfxr_initialize_for_dotnet_command_line");
     init_fptr = (hostfxr_initialize_for_runtime_config_fn)get_export(lib, "hostfxr_initialize_for_runtime_config");
     get_delegate_fptr = (hostfxr_get_runtime_delegate_fn)get_export(lib, "hostfxr_get_runtime_delegate");
     close_fptr = (hostfxr_close_fn)get_export(lib, "hostfxr_close");
 
-    assert(init_fptr && get_delegate_fptr && close_fptr);
+    assert(init_self_contained_fptr && init_fptr && get_delegate_fptr && close_fptr);
 }
 
 // Globals to hold runtime exports
 static load_assembly_and_get_function_pointer_fn get_managed_export_fptr;
 
-static void init_dotnet(const char_t* config_path)
+static void init_dotnet(const char_t* assembly_path)
 {
+    const char_t* config_path = NULL;
+    int rc;
+
+#ifdef DNNE_SELF_CONTAINED_RUNTIME
+    // Self-contained scenario support is experimental and relies upon the application scenario
+    // entry-point. The logic here is to trick the hosting API into initializing as an application
+    // but call the "load assembly and get delegate" instead of "run main". This has impact
+    // on the TPA make-up and hence the default ALC but does technically enable a self-contained
```

Comment:
> hence the default ALC 

Which means it impacts assembly loading in general, since ALCs all have a fallback of resolving through the default ALC. Not saying it is necessarily worth putting in the comment - just calling it out.